packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { ResidenceList } from '@deriv/api-types';
+import { Formik, Field, FieldProps } from 'formik';
+import { Autocomplete, DesktopWrapper, MobileWrapper, SelectNative, Loading } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { useWS } from '../../../../hooks';
+import { TReactChangeEvent } from 'Types';
+import { observer } from 'mobx-react';
+
+type TCountrySelectorProps = {
+    onSelect: (country: ResidenceList[number]) => void;
+    className: string;
+};
+
+type TValues = {
+    country_input: string;
+};
+
+const CountrySelector = ({ onSelect, className }: TCountrySelectorProps) => {
+    const WS = useWS('residence_list');
+    const { data: residence_list, is_loading } = WS;
+
+    const initial_form_values = {
+        country_input: '',
+    };
+
+    const validateFields = (values: TValues) => {
+        const errors: TValues = {
+            country_input: '',
+        };
+        const { country_input } = values;
+
+        if (!country_input) {
+            errors.country_input = localize('Please select the country of document issuance.');
+        } else if (!residence_list?.find(c => c.text === country_input)) {
+            errors.country_input = localize('Please select a valid country of document issuance.');
+        }
+
+        return errors;
+    };
+
+    const submitCountry = (values: TValues) => {
+        const matching_country = residence_list?.find(c => c.text === values.country_input);
+        if (matching_country) {
+            onSelect(matching_country);
+        }
+    };
+
+    React.useEffect(() => {
+        if (!residence_list) {
+            WS.send();
+        }
+    }, [WS, residence_list]);
+
+    return (
+        <Formik initialValues={initial_form_values} validate={validateFields} onSubmit={submitCountry}>
+            {({ errors, handleBlur, handleChange, setFieldValue, touched, values }) => (
+                <div className={className}>
+                    {!residence_list && is_loading && <Loading is_fullscreen={false} />}
```

Comment:
@hamid-deriv Just a question, Is there a reason to render the form itself while we just wanna show a loading state? 🤔

Reply:
@farzin-deriv Yes, because of the style of the `Loading` component, I need to wrap the `Loading` inside the `Formik` component to show the loading in the middle of the form. Otherwise, I need to change the styles of the Loading.